src/csharp/Grpc.Core.Tests/ChannelConnectivityTest.cs

```diff
@@ -73,6 +73,24 @@ public async Task Channel_WaitForStateChangedAsync()
             Assert.AreEqual(ChannelState.Ready, channel.State);
         }
 
+        [Test]
+        public async Task Channel_TryWaitForStateChangedAsync()
+        {
+            helper.UnaryHandler = new UnaryServerMethod<string, string>((request, context) =>
+            {
+                return Task.FromResult(request);
+            });
+
+            Assert.IsFalse(await channel.TryWaitForStateChangedAsync(channel.State, DateTime.UtcNow.AddMilliseconds(10)));
```

Comment:
nit: might be good to shorten this timeout to zero, just to have one less arbitrary delay (and make the intended behavior a bit more obvious).

Reply:
done.